src/Compilers/CSharp/Test/Semantic/Semantics/ScriptSemanticsTests.cs

```diff
@@ -958,5 +958,178 @@ public void FixedBuffer_02()
+
+            compilation.VerifyDiagnostics(
+                // (1,1): error CS7006: Expressions and statements can only occur in a method body
+                // System.Console.WriteLine(out var x, x);
+                Diagnostic(ErrorCode.ERR_GlobalStatement, "System.Console.WriteLine(out var x, x);").WithLocation(1, 1),
+                // error CS5001: Program does not contain a static 'Main' method suitable for an entry point
+                Diagnostic(ErrorCode.ERR_NoEntryPoint).WithLocation(1, 1)
+                );
+
+            syntaxTree = SyntaxFactory.ParseSyntaxTree(code, options: new CSharp.CSharpParseOptions(kind: SourceCodeKind.Script));
+            compilation = CreateCompilationWithMscorlib45(new[] { syntaxTree });
+            semanticModel = compilation.GetSemanticModel(syntaxTree, true);
+            node5 = ErrorTestsGetNode(syntaxTree);
+            Assert.Equal("WriteLine", node5.Name.ToString());
+            Assert.Null(semanticModel.GetSymbolInfo(node5.Name).Symbol);
+            x = syntaxTree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(id => id.Identifier.ValueText == "x").Single();
+            Assert.Equal("var Script.x", semanticModel.GetSymbolInfo(x).Symbol.ToTestDisplayString());
+
+            compilation.VerifyDiagnostics(
+                // (1,34): error CS7019: Type of 'x' cannot be inferred since its initializer directly or indirectly refers to the definition.
+                // System.Console.WriteLine(out var x, x);
+                Diagnostic(ErrorCode.ERR_RecursivelyTypedVariable, "x").WithArguments("x").WithLocation(1, 34)
+                );
+
+            syntaxTree = SyntaxFactory.ParseSyntaxTree(code, options: new CSharp.CSharpParseOptions(kind: SourceCodeKind.Script));
+            compilation = CreateCompilationWithMscorlib45(new[] { syntaxTree }, options: TestOptions.ReleaseExe.WithScriptClassName("Script1"));
+            semanticModel = compilation.GetSemanticModel(syntaxTree, true);
+            node5 = ErrorTestsGetNode(syntaxTree);
+            Assert.Equal("WriteLine", node5.Name.ToString());
+            Assert.Null(semanticModel.GetSymbolInfo(node5.Name).Symbol);
+            x = syntaxTree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(id => id.Identifier.ValueText == "x").Single();
+            Assert.Equal("var Script1.x", semanticModel.GetSymbolInfo(x).Symbol.ToTestDisplayString());
+
+            compilation.VerifyDiagnostics(
+                // (1,34): error CS7019: Type of 'x' cannot be inferred since its initializer directly or indirectly refers to the definition.
+                // System.Console.WriteLine(out var x, x);
+                Diagnostic(ErrorCode.ERR_RecursivelyTypedVariable, "x").WithArguments("x").WithLocation(1, 34)
+                );
+        }
+
+        [Fact]
+        [WorkItem(10023, "https://github.com/dotnet/roslyn/issues/10023")]
+        public void Errors_03()
```

Comment:
I guess the test is related to multiple features, it can go to either place and I think it is more appropriate here because it makes self-explanatory why we are not testing trees with this SourceCodeKind.